### PR TITLE
Bump mobx minor, mobx-react major versions

### PR DESCRIPTION
**Pull request by soyarsauce**
### What this PR does

Addresses requirement for some new syntax/approaches to https://github.com/TerriaJS/terriajs/issues/4183 and further UI development, by updating mobx versions - mainly hooks 

### Checklist

-   [x] unit tests still pass locally
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

**Comment by soyarsauce** [comment]
(worth noting that "unit tests pass locally" is laughable as probably 90% of the mobx tests don't cover those old ui ones! but that is being improved on with each new ui development pr)

File: lib/ReactViews/Map/Legend/DistanceLegend.jsx [hunk 1]
```diff
@@ -48,179 +47,173 @@ const distances = [
...
+      <div className={Styles.distanceLegend}>
+        <label>{this.state.distanceLabel}</label>
+        <div className={Styles.bar} style={barStyle} />
+      </div>
+    ) : null;
+
+    return distanceLabel;
+  }
+}
 module.exports = DistanceLegend;
```
Comment (tephenavies) [inline lib/ReactViews/Map/Legend/DistanceLegend.jsx:219]: Did you change this? Or just the spacing?

Comment (tephenavies) [reply lib/ReactViews/Map/Legend/DistanceLegend.jsx:219]: It's just spacing and minimal createClass -> class changes (according to `git diff --ignore-space-change origin/mobx origin/mobx-update-versions -- lib/ReactViews/Map/Legend/DistanceLegend.jsx`)

Comment (soyarsauce) [reply lib/ReactViews/Map/Legend/DistanceLegend.jsx:219]: Yeah just a little!

Comment (soyarsauce) [reply lib/ReactViews/Map/Legend/DistanceLegend.jsx:219]: You can't use the function version of `disposeOnUnmount` on a `createClass` - needs to be a class component

**Pull request merged**